### ISSUE TITLE
Label numbers generated by flow_from_directory

### ISSUE THREAD

**Issue by fabioperez** [fchollet/keras#3054]
I'm using [`flow_from_directory`](http://keras.io/preprocessing/image/) to get the training set from a folder with the following structure:

```
train
  class1
  class2
  class3
  ...
```

The generator is called as it follows:

```
train_generator = train_datagen.flow_from_directory( 
        train_data_dir,                              
        target_size=(img_height, img_width),         
        batch_size=32,                               
        class_mode='categorical')                    
```

I am not setting the argument `classes`, but I was expecting to get the labels in alphabetical order.

> **classes**: optional list of class subdirectories (e.g. `['dogs', 'cats']`). Default: None. If not provided, the list of classes will be automatically inferred (and the order of the classes, which will map to the label indices, will be alphanumeric).

However, when I classify the training images (for checking which labels are being returned), I'm don't get any specific ordering. The training goes well (accuracy of ~85%), and there is a consistency with the output labels when classifying images from the same class.

How can I infer the labels numbers generated by `flow_from_directory` and map them to the classes?

**Comment by fabioperez** [comment]
This was fixed by https://github.com/fchollet/keras/pull/3052.

**Comment by sunyongke** [comment]
how to resolve ?

**Comment by fabioperez** [comment]
Just use the most recent version of Keras.

**Comment by sunyongke** [comment]
Yes,I have resoved. Thanks

2017年1月25日 06:04，"Fábio Perez" <notifications@github.com>写道：

**Comment by StuvX** [comment]
how to resolve?
